src/pipelinedefinition.hpp

```diff
@@ -98,7 +98,7 @@ class PipelineDefinition {
     void makeSubscriptions(ModelManager& manager);
     void resetSubscriptions(ModelManager& manager);
 
-    Status getInputsInfo(tensor_map_t& inputsInfo, const ModelManager& manager) const;
-    Status getOutputsInfo(tensor_map_t& outputsInfo, const ModelManager& manager) const;
+    virtual Status getInputsInfo(tensor_map_t& inputsInfo, const ModelManager& manager) const;
```

Comment:
Just a clarifying question:
Is omitting making the destructor virtual intentional here?
I mean with the code as it stands it is correct since you are not handling anywhere pointers to base/derived class.
However if you start adding containers of derived pipeline definitions this may be issue if those are kept as a container of pointers of some kind.